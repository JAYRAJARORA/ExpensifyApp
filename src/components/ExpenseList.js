import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => (
                    <ExpenseListItem key={expense.id} {...expense}/>
                ))
            )
        }
    </div>
);

// mapping redux store state to react component props
// when store changes this is automatically going to rerun bringing in fresh values
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
    };
};

// an HOC
export default connect(mapStateToProps)(ExpenseList); 