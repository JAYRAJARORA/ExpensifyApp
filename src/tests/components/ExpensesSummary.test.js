import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import { shallow} from 'enzyme';

test('should render expenses summary correctly fro 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={5.34}/>)
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary correctly for 2 expenses', () => {
    const wrapper = shallow(<ExpensesSummary  expenseCount={4} expensesTotal={5455.55}/>)
    expect(wrapper).toMatchSnapshot();
});