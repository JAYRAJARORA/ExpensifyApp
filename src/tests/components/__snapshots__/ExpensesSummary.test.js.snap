// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary correctly for 2 expenses 1`] = `
<div>
  <p>
    Viewing 
    4
     
    expenses
     with amount 
    $54.56
  </p>
</div>
`;

exports[`should render expenses summary correctly fro 1 expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     with amount 
    $0.05
  </p>
</div>
`;
